src/components/ProjectDomain/projectdomainTable.tsx

```diff
@@ -95,17 +86,15 @@ const ProjectDomainTable: React.FC = () => {
     }
   };
 
-  const handleSwitchChange = async (labelId: string, checked: boolean) => {
-    // Initialize toast
-    const updatedDomainDict = { ...domainDataDict };
-    if (updatedDomainDict[labelId]) {
-      updatedDomainDict[labelId].status = checked
+  const handleSwitchChange = async (
+    labelId: string,
+    checked: boolean,
+    index: number,
+  ) => {
+    try {
+      domainData[index].status = checked
```

Comment:
Although this is technically correct and will update the state value, but you should never mutate a state object directly, always treat state object as read only variables.
Use setter function to update new value to state